### PR TITLE
[Feature]: add unpacker and views for codabench project

### PR DESCRIPTION
**Pull request by AuroraLauu**
# @ mention of reviewers
Code: @ckcollab 
Hand test: @dde6khkg 

# A brief description of the purpose of the changes contained in this PR.
1. add Unpacker「CodabenchV1Unpacker」support  Benchmark Creation by Bundle Upload and its correspond unit test「test_codabench_v1_unpacker」

2. add type('competition' or 'benchmark') in Competition Model in order to distinguish this is "Competition" or "Benchmark"

3. register new app 'benchmark' in base.py settings

4. add new field 'type' in 'CompetitionSerializer', 'CompetitionDetailSerializer', 'CompetitionSerializerSimple'

5. add benchmark relevant views and urls, this part of the code reuses code from the Competition section
- we just added the routing, In particular 
  - for incoming requests from the benchmark view, the rule is type='bench'
  - for incoming requests from the competition view, the rule is type='competition'.

...

# Issues this PR resolves
...

# Known issues to be addressed in a separate PR
...

# A checklist for hand testing
- [x] codabench v1 unpacker test

# Any relevant files for testing
[link]('#') to any relevant files (or drag and drop into github)

# Misc. comments
...

# Checklist
- [x] Code review by me 
- [x] Hand tested by me 
- [x] I'm proud of my work
- [x] Code review by reviewer
- [x] Hand tested by reviewer
- [x] Ready to merge

### CODE REVIEW

File: src/apps/competitions/unpackers/codabench_v1.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+
+from django.utils import timezone
+
+from competitions.unpackers.v2 import V2Unpacker
+from .utils import CompetitionUnpackingException
+from ..models import Competition
+
+
+class CodabenchV1Unpacker(V2Unpacker):
```
Comment (ckcollab) [inline src/apps/competitions/unpackers/codabench_v1.py:9]: I don't think we need a whole new unpacker for codabench? Will have to discuss with Isabelle a bit more?

Comment (AuroraLauu) [reply src/apps/competitions/unpackers/codabench_v1.py:9]: okay, that's a good idea

File: src/apps/competitions/models.py [hunk 2]
```diff
@@ -35,8 +43,11 @@ class Competition(ChaHubSaveMixin, models.Model):
 
     queue = models.ForeignKey('queues.Queue', on_delete=models.SET_NULL, null=True, blank=True, related_name='competitions')
 
+    # we use filed type to distinguish 'competition' and 'benchmark'
+    type = models.CharField(max_length=128, choices=COMPETITION_TYPE, default=COMPETITION, null=True, blank=True)
```
[the remark this answers is not in the document]
Comment (AuroraLauu) [reply src/apps/competitions/models.py:47]: yep, i will change this variable name, and default should be COMPETITION,

**Comment by AuroraLauu** [comment]
Hi @ckcollab , according to your suggestion, I have updated the type field to competition_type and set it to not be empty, the default value is competition.
Also, if we don't need codabench unpacker, then the new benchmark yaml will be deprecated and the existing competition yaml will be reused completely.

In the new PR I submitted 2 new files.
1. urls.py
2. views.py
Both files are related to Benchmark, but with routing added to the original Competition.

File: src/apps/api/urls.py [hunk 3]
```diff
@@ -6,10 +6,12 @@
 from rest_framework.permissions import AllowAny
 
 from api.views.competitions import front_page_competitions
-from .views import analytics, competitions, datasets, profiles, leaderboards, submissions, tasks, queues
+from .views import analytics, competitions, datasets, profiles, leaderboards, submissions, tasks, queues, benchmarks
 
 
 router = SimpleRouter()
+router.register('benchmarks', benchmarks.BenchmarkViewSet)
+router.register('benchmarks_status', benchmarks.BenchmarkCreationTaskStatusViewSet)
```
Comment (ckcollab) [inline src/apps/api/urls.py:14]: why don't we keep existing viewsets and pass a flag/argument to filter by `competition_type`?

Comment (AuroraLauu) [reply src/apps/api/urls.py:14]: wow, that will be great, need to pass a flag/argument from front end side

Comment (ckcollab) [reply src/apps/api/urls.py:14]: :)

File: src/apps/competitions/models.py [hunk 4]
```diff
@@ -35,8 +43,11 @@ class Competition(ChaHubSaveMixin, models.Model):
 
     queue = models.ForeignKey('queues.Queue', on_delete=models.SET_NULL, null=True, blank=True, related_name='competitions')
 
+    # we use filed type to distinguish 'competition' and 'benchmark'
```
Comment (ckcollab) [inline src/apps/competitions/models.py:46]: :+1:

**Comment by AuroraLauu** [comment]
Hello @ckcollab , I've changed the PR according to what was discussed at the meeting. can you help me review it?

**Comment by dde6khkg** [comment]
It looks like you might need to migrate and push that migration, but other than that it works fine for me.

**Comment by AuroraLauu** [comment]
> It looks like you might need to make migrations and push that migration, but other than that it works for me!

Hi Nic，i already add a migration files into it. Thanks for the reminder.

**Comment by dde6khkg** [comment]
Nice! Looks like it works!

**Pull request merged**